### DISCUSSION TITLE
play recorded video from 3rd party Windows program

### DISCUSSION THREAD

**Discussion by eliangbb** [AxisCommunications/discussions#116] [Axis Windows SDKs] [answered]
Hi there, 
I am trying to play recorded Axis camera video from my Windows program. Is there an Axis module/program that we can call with timestamp from our program to bring up something like below as what is under Axis Camera Station for playback as below:  (BTW, is there a sample program for remote video playback ONLY?). Thank you very much.
![AxisPlayback](https://user-images.githubusercontent.com/108364188/202032855-e37cc9a8-38cc-4528-bd89-03aab153b629.png)

**Comment by mattias-kindborg-at-work** [comment]
Hi @eliangbb. For live video we have [Media Stream Library JS](https://github.com/AxisCommunications/media-stream-library-js) but when it comes to playback of recorded video it often has to do with how that video is stored, which makes it a harder question to answer. When you mentioned AXIS Camera Station, did you also mean that you would like to playback recordings from AXIS Camera Station, or just that you where using AXIS Camera Station as an example?

**Reply by eliangbb** [reply]
Hi Mattias, 
Thank you so much. We would like to play recorded video in our program, not live video. What I mentioned for Axis Camera station playback function (as screenshot shows) is just an example, we would like to make our program show something like this including playing video, with time slider, all the buttons (play/pause, etc...) and other options.
So, about playback of recorded video, what do you have? And what is the better and simplest way to stored the video for this? Thank you again.

**Reply by eliangbb** [reply]
Hi Mattias, one more thing, our program will just play the recorded video ONLY, no other functions. It will use timestamp and camera info to play/retrieve the matched video. Thanks.

**Reply by eliangbb** [reply]
Hi Mattias,
More about our situation: 
could we have a **_Client Player module_** that plays back recorded video from Axis Camera Station VMS? That module is AxisFilePlayer.exe, right(check screen below)? Or, it is something else we can use in our program. If we have this module to run in our own program, it will solve our problem. 
![player](https://user-images.githubusercontent.com/108364188/202568108-85668490-6204-46ea-ace7-ee1154064b78.png)

**Reply by eliangbb** [reply]
As I mentioned, we need to use timestamp and camera info to bring up and play the matched video automatically. We don't want user to select any files manuall and we need to bypass this file selection part.

**Reply by mattias-kindborg-at-work** [reply]
Hi @eliangbb. Unfortunately I don't know of any module that you can use in your application, it's not a problem we've built open source for. I would also refrain from borrowing applications and libraries from AXIS Camera Station. It's a product based on a proprietary codebase and not licensed to be shared.

**Reply by haichaoji123** [reply]
if you want to integrate with ACS into your application, you can apply for ACS API,  with that API, you can implement the function as you described. Your company need register as the TIP, https://www.axis.com/partner/technology-integration-partner-program
 and then you can request the ACS API.

**Reply by eliangbb** [reply]
Hi @haichaoji123. Thank you so much. I have two more questions about using ACS API for playback.

ACS API for Playback works fine under ACS machine web browser if I use "localhost" as server address, and the video file can be downloaded into Windows "Downloads" folder. But if I do this from remote machine and use ACS machine external IP (see example below), it pops up to ask ACS machine user/password for the **_authentication_** (screen is below):
" _**https://108.15.64.2:55756/Acs/Streaming/Video/Playback/Mp4/?camera=173_72ccb9bb-5eb8-4982-8f52-0bf04ff4ee0f&quality=medium&start=2022-12-14-211210-0000000Z&end=2022-12-14-211410-0000000Z&audio=0**_ "
![acsLogin](https://user-images.githubusercontent.com/108364188/208278610-150b4556-8aaf-4125-8000-7ec090d8122f.png)

**So here are my questions:**
1. What should be done on ACS server or machine to skip authentication part while doing "https://...." (as above URI) under remote web browser?
2. If I use Python **requests** for http post for the above URI, what should I do to make video file downloaded just like web browser? How about the authentication part? The reason I ask this is that we need to retrieve the video file from ACS by running our own program. 

thank you again,

**Reply by apankina** [reply]
Thank you for also posting your question in our development helpdesk where  you have a response. 
In general, please review the documentation for details on authentication and make sure you manage the server certificate within your code.

**Reply by eliangbb** [reply]
thank you!!! your sample playback c# code shown in your development help desk works for us.

**Comment by LenaMagnusson** [comment, marked as answer]
The API for AXIS Camera Station is still in active development with frequent changes. We are only able to support a limited number of developers and can thus only make it available for a small number of developers. If you are interested in developing an integration to AXIS Camera Station and want access to the ACS API, please contact your local Axis representative.

**Reply by eliangbb** [reply]
Hi @LenaMagnusson, thank you, very helpful. I am trying and asked more questions above.
